runtime/src/main/java/org/corfudb/runtime/collections/Table.java

```diff
@@ -58,18 +58,35 @@
     @Getter
     private final TableMetrics metrics;
 
+    @Getter
+    private final Class<K> keyClass;
+
+    @Getter
+    private final Class<V> valueClass;
+
+    @Getter
+    private final Class<M> metadataClass;
+
     /**
      * Returns a Table instance backed by a CorfuTable.
      *
-     * @param namespace               Namespace of the table.
-     * @param fullyQualifiedTableName Fully qualified table name.
-     * @param valueSchema             Value schema to identify secondary keys.
-     * @param corfuRuntime            Connected instance of the Corfu Runtime.
-     * @param serializer              Protobuf Serializer.
+     * @param namespace               namespace of the table
+     * @param fullyQualifiedTableName Fully qualified table name
+     * @param kClass                  key class
+     * @param vClass                  value class
+     * @param mClass                  metadata class
+     * @param valueSchema             value schema to identify secondary keys
+     * @param corfuRuntime            connected instance of the Corfu Runtime
+     * @param serializer              protobuf serializer
+     * @param streamingMapSupplier    supplier of underlying map data structure
+     * @param versionPolicy           versioning policy
      */
     @Nonnull
     public Table(@Nonnull final String namespace,
                  @Nonnull final String fullyQualifiedTableName,
+                 @Nonnull final Class<K> kClass,
```

Comment:
why not pass `TableSchema` instead of these 3 parameters?

Reply:
These three parameters are same as what were passed in from the caller.